stingray/pulse/pulsar.py

```diff
@@ -303,7 +304,107 @@ def profile_stat(profile, err=None):
     return np.sum((profile - mean) ** 2 / err ** 2)
 
 
-def z_n(phase, n=2, norm=1):
+@functools.lru_cache(maxsize=128)
```

Comment:
I'm curious what this does?

Reply:
It caches the operations. The method creates an array of cos / sin values corresponding to the phase of each of the n bins of a profile. Thanks to that decorator, the results are cached and the operation done only once